src/libs/ReportUtils.js

```diff
@@ -1661,14 +1672,25 @@ function getTransactionsWithReceipts(iouReportID) {
  * @returns {Boolean}
  */
 function areAllRequestsBeingSmartScanned(iouReportID, reportPreviewAction) {
-    const transactionsWithReceipts = getTransactionsWithReceipts(iouReportID);
+    const transactionsWithReceipts = getDistanceTransactions(iouReportID);
     // If we have more requests than requests with receipts, we have some manual requests
     if (ReportActionsUtils.getNumberOfMoneyRequests(reportPreviewAction) > transactionsWithReceipts.length) {
         return false;
     }
     return _.all(transactionsWithReceipts, (transaction) => TransactionUtils.isReceiptBeingScanned(transaction));
 }
 
+/**
+ * Check if any of the distance transactions in the report has invalid route
+ *
+ * @param {Object|null} iouReportID
+ * @returns {Boolean}
+ */
+function hasDistanceTransactionRouteErrors(iouReportID) {
+    const distanceTransactions = getDistanceTransactions(iouReportID);
```

Comment:
It would be more efficient to check if it's a distance transaction within the some callback, that way we don't have to loop through all transactions once we find one with a route error. Also I think we can just check if it has a route error since only distance transactions can.

Reply:
Let's update both functions to be more performant. I don't see why we should purposefully do more work than necessary.

We can just do
```js
return _.some(allTransactions, (transaction) => TransactionUtils.hasRouteError(transaction));
```